posts/Stef/2019-11-18-hibernate-orm-with-panache-in-quarkus.adoc

```diff
@@ -0,0 +1,462 @@
+
+This allows many simple queries to be simplified to a minimum, while allowing complex queries to be left as-is.
+
+Now, our REST endpoint is not changed much, but let's include it for good measure:
+
+[source%nowrap,java]
+----
+@Path("/")
+public class PersonEndpoint {
+    @Inject
+    private PersonDao personDao;
+    
+    @GET
+    @Path("people")
+    public List<Person> all(){
+        return personDao.findAll().list();
+    }
+
+    @GET
+    @Path("people/by-name")
+    public List<Person> findByName(@PathParam String name){
+        return personDao.findByName(name);
+    }
+
+    @GET
+    @Path("people/born-after")
+    public List<Person> findBornAfter(@PathParam Date date){
+        return personDao.findBornAfter(date);
+    }
+
+    @GET
+    @Path("person/{id}")
+    public Person findById(@PathParam Long id){
+        Person p = personDao.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        return p;
+    }
+
+    @PUT
+    @Path("person/{id}")
+    public void updatePerson(@PathParam Long id, Person newPerson){
+        Person p = personDao.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        p.birth = newPerson.birth;
+        p.firstName = newPerson.firstName;
+        p.lastName = newPerson.lastName;
+    }
+
+    @DELETE
+    @Path("person/{id}")
+    public void deletePerson(@PathParam Long id){
+        Person p = personDao.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        personDao.delete(p);
+    }
+    
+    @POST
+    @Path("people")
+    public Response newPerson(@Context UriInfo uriInfo, Person newPerson){
+        Person p = new Person();
+        p.birth = newPerson.birth;
+        p.firstName = newPerson.firstName;
+        p.lastName = newPerson.lastName;
+        personDao.persist(p);
+        
+        URI uri = uriInfo.getAbsolutePathBuilder()
+                .path(PersonEndpoint.class)
+                .path(PersonEndpoint.class, "findById")
+                .build(p.id);
+        return Response.created(uri).build();
+    }
+}
+----
+
+This is already a long way towards reducing boilerplate, don't you think?
+
+== Going the extra mile and getting rid of the DAO
+
+Data Access Objects are mostly useful when you have one or more of the following situations:
+
+- The entity type is shared between projects written for different stacks. One project will use DAOs written for WildFly, another for Spring.
+- The entity type is shared between projects written for different use-cases. One project will handle the entity in one way, while another will differ entirely.
+- You need to mock your DAOs in tests.
+- Your entity type is crammed so full of getters and setters that adding any model method will exceed the maximum method count.
+
+While the last reason is a joke, the first reasons are entirely reasonable.
```

Comment:
"While the last reason started as a joke, we as human have the tendency to split things when they are getting too big. A Class with tons of getters/setters makes us reluctant to add more methods.

Reply:
Thanks.